nx/lib/nx.ex

```diff
@@ -897,6 +897,16 @@ defmodule Nx do
     %T{shape: shape, type: type, names: names, data: %Nx.TemplateBackend{}}
   end
 
+  for t <- [:u8, :u16, :u32, :u64, :s8, :s16, :s32, :s64, :bf16, :f16, :f32, :f64, :c64, :c128] do
```

Comment:
I would skip the complex ones for now because maybe for them `c64(real, img)` is better? Anyway, we can discuss this later and the below is enough to get started.

```suggestion
  for t <- [:u8, :u16, :u32, :u64, :s8, :s16, :s32, :s64, :bf16, :f16, :f32, :f64] do
```

Reply:
Makes sense. We already have Nx.complex. Worst case scenario people would be doing `complex(f32(real), f32(imag))` or `f32(real) + Nx.Constants.i() * f32(imag)` or variations on that